### PR TITLE
go 1.15.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/go.rb [hunk 1]
```diff
@@ -4,9 +4,9 @@ class Go < Formula
   license "BSD-3-Clause"
 
   stable do
-    url "https://golang.org/dl/go1.15.src.tar.gz"
-    mirror "https://fossies.org/linux/misc/go1.15.src.tar.gz"
-    sha256 "69438f7ed4f532154ffaf878f3dfd83747e7a00b70b3556eddabf7aaee28ac3a"
+    url "https://dl.google.com/go/go1.15.1.src.tar.gz"
```
Comment (SMillerDev) [inline Formula/go.rb:7]:
```suggestion
    url "https://golang.org/dl/go1.15.1.src.tar.gz"
```

Comment (chenrui333) [reply Formula/go.rb:7]: `https://dl.google.com/go/go1.15.src.tar.gz` is the true location.

```
$ curl -I https://golang.org/dl/go1.15.src.tar.gz
HTTP/1.1 302 Found
Date: Tue, 01 Sep 2020 19:23:36 GMT
Content-Type: text/html; charset=utf-8
Location: https://dl.google.com/go/go1.15.src.tar.gz
```

Comment (chenrui333) [reply Formula/go.rb:7]: Similar work that gradle team has done, https://github.com/gradle/gradle/pull/2152

Comment (chenrui333) [reply Formula/go.rb]: Revert the style change, we can talk about it in a separate thread.

**Comment by mojotx** [comment]
Out of curiosity, is there anything blocking this PR? [Go 1.15.1](https://github.com/golang/go/releases/tag/go1.15.1) is a security fix for https://github.com/golang/go/issues/40928.

**Review by SMillerDev — DISMISSED** [review]

**Comment by SMillerDev** [comment]
> Out of curiosity, is there anything blocking this PR?

The failing formula are, they need to be checked and fixed.

**Comment by mojotx** [comment]
> > Out of curiosity, is there anything blocking this PR?
> 
> The failing formula are, they need to be checked and fixed.

Okay, it looks like a dependency issue with the [charm](https://github.com/Homebrew/homebrew-core/blob/master/Formula/charm.rb) recipe. If nobody beats me to it, I will dig into it later.  

```
==> go build github.com/juju/charmstore-client/cmd/charm
Last 150 lines from /Users/brew/actions-runner/_work/homebrew-core/homebrew-core/bottles/logs/charm/03.go:
2020-09-01 22:18:48 +0100

go
build
github.com/juju/charmstore-client/cmd/charm

go: gopkg.in/juju/charmstore.v5@v5.6.4 requires
	github.com/CanonicalLtd/omniutils@v0.0.0-20180607151646-ed3ed7dca482: invalid version: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /private/tmp/charm-20200901-29376-1iaj8qk/charmstore-client-2.4.0/pkg/mod/cache/vcs/05d357a6fb7db8f493808e9721a54716ec1be125069f304bb38b18b745c601ec: exit status 128:
	fatal: could not read Username for 'https://github.com': terminal prompts disabled

READ THIS: https://docs.brew.sh/Troubleshooting
```

**Comment by chenrui333** [comment]
@ydnar that is true, but let's minimize the impact and the disruption. (I am going to see what I can do really quick)

**Comment by ydnar** [comment]
@chenrui333 anyway to record which failures aren’t regressions?

Maybe mark which tests are known failures, so they can be ignored if they fail in subsequent builds (or re-fail if they pass)?

**Comment by chenrui333** [comment]
yeah, the merge decision is usually ad-hoc as well, so knowing the previous failures is actually pretty trivial.

**Comment by ydnar** [comment]
Aye. It's more the time. The smoke tests take long enough, and then there's the inevitable back and forth discussion while Go dev moves forward. 1.15.2 is almost ready.

My opinion: Ad-hoc process shouldn't block security updates. For major/minor updates, sure.

**Comment by chenrui333** [comment]
> Aye. It's more the time. The smoke tests take long enough, and then there's the inevitable back and forth discussion while Go dev moves forward. 1.15.2 is almost ready.
> 
> My opinion: Ad-hoc process shouldn't block security updates. For major/minor updates, sure.

Alright, let's merge this security fix, I will follow up with the regression testing failures. Appreciate your idea in here.

**Comment by chenrui333** [comment]
record the upgrade failure in [here](https://github.com/Homebrew/homebrew-core/issues/59698#issuecomment-687385975)

**Pull request closed without merging**